acstools/findsat_mrt.py

```diff
@@ -0,0 +1,1129 @@
+        None.
+
+        '''
+
+        
+        if root is None:
+            root = self.root
+        if output_dir is None:
+            output_dir = self.output_dir
+        if save_mrt is None:
+            save_mrt = self.save_mrt
+        if save_mask is None:
+            save_mask = self.save_mask
+        if save_diagnostic is None:
+            save_diagnostic = self.save_diagnostic
+        if save_catalog is None:
+            save_catalog = self.save_catalog
+            
+        if save_mrt == True:
+            if self.mrt is not None:
+                fits.writeto('{}/{}_mrt.fits'.format(output_dir,root),self.mrt,overwrite=True)
+            else:
+                LOG.error('No MRT to save')
+                
+        if save_mask == True:
+            if self.mask is not None:
+                hdu0 = fits.PrimaryHDU()
+                if self.header is not None:
+                    hdu0.header = self.header #copying over original image header
+                hdu1 = fits.ImageHDU(self.mask.astype(int))
+                hdu2 = fits.ImageHDU(self.segment.astype(int))
+                hdul = fits.HDUList([hdu0,hdu1])
+                hdul.writeto('{}/{}_mask.fits'.format(output_dir,root),overwrite=True)
+            else:
+                LOG.error('No mask to save')
+             
+        if save_diagnostic==True:
+            fig,[[ax1,ax2],[ax3,ax4]] = plt.subplots(2,2,figsize=(20,10))
+            self.plot_image(ax=ax1)
+            self.plot_mrt(ax=ax2)
+            self.plot_image(ax=ax3)
+            ax3.imshow(self.mask,alpha=0.5,origin='lower',aspect='auto',cmap='Reds')
+            ax3_xlim = ax3.get_xlim()
+            ax3_ylim = ax3.get_ylim()
+                            
+            self.plot_mrt(ax=ax4)
+            if self.source_list is not None:
+                for s in self.source_list:
+                    color='red'
+                    x1,y1 = s['endpoints'][0]
+                    x2,y2 = s['endpoints'][1]
+                    if (s['status'] == 2):
+                        color='turquoise'
+                    elif (s['status'] == 1):
+                        color='orange'
+                    ax3.plot([x1,x2],[y1,y2],color=color,lw=5,alpha=0.5)
+                    ax4.scatter(s['xcentroid'],s['ycentroid'],edgecolor=color,facecolor='none',s=100,lw=2)
+            #sometimes overplotting the "good" trails can cause axes to change
+            ax3.set_xlim(ax3_xlim)
+            ax3.set_ylim(ax3_ylim)
+            plt.tight_layout()
+            plt.savefig('{}/{}_diagnostic.png'.format(output_dir,root))
+            if self.plot==False:
+                plt.close()
+
+        if save_catalog==True:
+            if self.source_list is not None:
+                self.source_list.write('{}/{}_catalog.fits'.format(output_dir,root),overwrite=True)
+            else:
+                #create an empty catalog and write that. It helps to have this for future analysis purposes even if it's empty
```

Comment:
I let an empty catalog be written because it made it clear that the image was inspected but nothing found, as opposed to an image that was never inspected. If I did not write a catalog for images where no trails were found, I could make a note in the log, but I feel like this adds another layer of complexity to analysis downstream when merging all the catalog files. With that explanation, is it ok to leave the code as is? Happy to discuss alternatives!